### ISSUE TITLE
"Docker Desktop appears not to be running" but it is

### ISSUE THREAD

**Issue by seanf** [WhitewaterFoundry/Pengwin#471]
**Describe the bug**
According to the system tray icon, Docker Desktop is running, but `pengwin-setup` disagrees.

**To Reproduce**
1. Install Docker Desktop 2.0.0.3 (31259)
2. Restart when prompted to activate HyperV etc
3. Run pengwin-setup, select Tools then Docker, say Yes
4. See error: `Docker Desktop appears not to be running, please check it and ensure that it is running correctly. Would you like to try again?`

**Expected behavior**
Docker bridge should be installed

**Screenshots**

**Additional context**

**Basic Troubleshooting Checklist**

[x] I have searched Google for the error message.
[x] I have checked official WSL troubleshooting documentation: https://docs.microsoft.com/en-us/windows/wsl/troubleshooting#confirm-wsl-is-enabled.
[x] I have searched the official Microsoft WSL issues page: https://github.com/Microsoft/WSL/issues.
[x] I have searched the Pengwin issues page: https://github.com/WhitewaterFoundry/Pengwin/issues.
[ ] I have reset Pengwin: Settings->Apps->Apps & features->Pengwin->Advanced Options->Reset. 
[ ] I have disabled and re-enabled WSL in Windows Features.
[ ] I have run Windows 10 updates and restarted.

What other troubleshooting have you attempted?

Insert here:

**Pengwin Version**

Find: Settings->Apps->Apps & features->Pengwin->Advanced Options->Version.

Insert here: 1.2.4.0

**Windows Build**

Run 'systeminfo | findstr /C:"OS"' in Command Prompt and insert here:

OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.17763 N/A Build 17763
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Member Workstation
OS Build Type:             Multiprocessor Free
BIOS Version:              Dell Inc. 1.10.1, 2019-04-26

For help on retrieving: https://docs.microsoft.com/en-us/windows/wsl/troubleshooting#check-your-build-number

**Comment by crramirez** [comment]
Please run pengwin-setup with --debug and send us the output

**Comment by seanf** [comment]
Thanks @crramirez. The log is below. 
```
➜  ~ pengwin-setup --debug
Running in debug/verbose mode
++ shift
++ [[ 0 -gt 0 ]]
++ wslpath 'C:\'
+++ cmd-exe /c 'echo %HOMEDRIVE%%HOMEPATH%'
+++ tr -d '\r'
++ readonly 'wHomeWinPath=C:\Users\seanf'
++ wHomeWinPath='C:\Users\seanf'
+++ wslpath -u 'C:\Users\seanf'
++ readonly wHome=/mnt/c/Users/seanf
++ wHome=/mnt/c/Users/seanf
++ readonly CANCELLED=CANCELLED
++ CANCELLED=CANCELLED
++ SetupDir=/usr/local/pengwin-setup.d
++ readonly GOVERSION=1.12
++ GOVERSION=1.12
+ WelcomePrompt
+ whiptail --title 'Welcome to Pengwin' --msgbox 'Thank you for supporting sustainable independent open source development.\n
Pengwin comes with a core set of useful packages pre-installed, such as curl, git, and wslu.\n
pengwin-setup allows you to install additional hand-curated selections for Pengwin and configure optional WSL-specific settings. \n
Many more packages are available via the apt package manager and optional pip, npm, and RubyGems package managers.' 17 80
+ continue_prompt
+ confirm --title 'Continue setup?' --yesno 'Would you like to continue pengwin-setup?' 8 45
+ [[ ! -n '' ]]
+ whiptail --title 'Continue setup?' --yesno 'Would you like to continue pengwin-setup?' 8 45
+ return 0
+ echo 'Starting pengwin-setup.'
Starting pengwin-setup.
+ [[ ! -n '' ]]
+ CheckUpgrades
+ echo 'Updating package database'
Updating package database
+ sudo apt-get update
[sudo] password for seanf:
Hit:1 https://packagecloud.io/whitewaterfoundry/pengwin-base/debian buster InRelease
Hit:4 https://packagecloud.io/whitewaterfoundry/pengwin-setup/debian buster InRelease
Get:2 https://cdn-aws.deb.debian.org/debian testing InRelease [163 kB]
Hit:7 https://packagecloud.io/whitewaterfoundry/wslu/debian buster InRelease
Get:3 https://cdn-aws.deb.debian.org/debian testing-updates InRelease [46.8 kB]
Get:5 https://cdn-aws.deb.debian.org/debian-security testing/updates InRelease [39.1 kB]
Get:6 https://cdn-aws.deb.debian.org/debian unstable InRelease [210 kB]
Get:8 https://cdn-aws.deb.debian.org/debian testing/main amd64 Packages.diff/Index [27.9 kB]
Get:9 https://cdn-aws.deb.debian.org/debian testing/main amd64 Packages 2019-06-06-2122.13.pdiff [1,444 B]
Get:9 https://cdn-aws.deb.debian.org/debian testing/main amd64 Packages 2019-06-06-2122.13.pdiff [1,444 B]
Get:10 https://cdn-aws.deb.debian.org/debian unstable/main amd64 Packages.diff/Index [27.9 kB]
Get:11 https://cdn-aws.deb.debian.org/debian unstable/non-free amd64 Packages.diff/Index [27.8 kB]
Get:12 https://cdn-aws.deb.debian.org/debian unstable/main amd64 Packages 2019-06-06-0813.50.pdiff [936 B]
Get:13 https://cdn-aws.deb.debian.org/debian unstable/main amd64 Packages 2019-06-06-1410.53.pdiff [2,838 B]
Get:14 https://cdn-aws.deb.debian.org/debian unstable/main amd64 Packages 2019-06-06-2122.13.pdiff [723 B]
Get:14 https://cdn-aws.deb.debian.org/debian unstable/main amd64 Packages 2019-06-06-2122.13.pdiff [723 B]
Get:15 https://cdn-aws.deb.debian.org/debian unstable/non-free amd64 Packages 2019-06-06-1410.53.pdiff [412 B]
Get:15 https://cdn-aws.deb.debian.org/debian unstable/non-free amd64 Packages 2019-06-06-1410.53.pdiff [412 B]
Fetched 549 kB in 12s (46.0 kB/s)
Reading package lists... Done
+ '[' -f /etc/apt/.dist-upgrade ']'
+ echo 'Running upgrade check...'
Running upgrade check...
++ sudo apt-get upgrade --show-upgraded --assume-no
++ grep pengwin
+ UPGRD_CHECK=
+ [[ '' == *\p\e\n\g\w\i\n* ]]
+ echo 'Holding pengwin-base & pengwin-setup to ensure no changes while operating'
Holding pengwin-base & pengwin-setup to ensure no changes while operating
+ sudo apt-mark hold pengwin-base pengwin-setup
+ install_menu --debug
++ menu --title pengwin-setup --checklist --separate-output '\nHand-curated add-ons [SPACE to select, ENTER to confirm]:' 15 99 7 EDITORS 'Install text editors neovim, emacs, or Visual Studio Code (requires X)    ' off GUI 'Install an X server or various other GUI applications' off MAINTENANCE 'Various maintenance tasks like home backup' off PROGRAMMING 'Install various programming languages support' off SERVICES 'Enable services support (SSH, rc.local)' off SETTINGS 'Change various settings in Pengwin' off TOOLS 'Install applications or servers' off
++ local menu_choice
+++ whiptail --title pengwin-setup --checklist --separate-output '\nHand-curated add-ons [SPACE to select, ENTER to confirm]:' 15 99 7 EDITORS 'Install text editors neovim, emacs, or Visual Studio Code (requires X)    ' off GUI 'Install an X server or various other GUI applications' off MAINTENANCE 'Various maintenance tasks like home backup' off PROGRAMMING 'Install various programming languages support' off SERVICES 'Enable services support (SSH, rc.local)' off SETTINGS 'Change various settings in Pengwin' off TOOLS 'Install applications or servers' off
++ menu_choice=TOOLS
++ local exit_status=0
++ [[ 0 != 0 ]]
++ [[ -z TOOLS ]]
++ echo TOOLS
+ local menu_choice=TOOLS
+ echo Selected: TOOLS
Selected: TOOLS
+ [[ TOOLS == \C\A\N\C\E\L\L\E\D ]]
+ local exit_status
+ [[ TOOLS == *\E\D\I\T\O\R\S* ]]
+ [[ TOOLS == *\G\U\I* ]]
+ [[ TOOLS == *\M\A\I\N\T\E\N\A\N\C\E* ]]
+ [[ TOOLS == *\P\R\O\G\R\A\M\M\I\N\G* ]]
+ [[ TOOLS == *\S\E\R\V\I\C\E\S* ]]
+ [[ TOOLS == *\S\E\T\T\I\N\G\S* ]]
+ [[ TOOLS == *\T\O\O\L\S* ]]
+ echo TOOLS
TOOLS
+ bash /usr/local/pengwin-setup.d/tools.sh --debug
Running in debug/verbose mode
++ shift
++ [[ 0 -gt 0 ]]
++ wslpath 'C:\'
+++ cmd-exe /c 'echo %HOMEDRIVE%%HOMEPATH%'
+++ tr -d '\r'
++ readonly 'wHomeWinPath=C:\Users\seanf'
++ wHomeWinPath='C:\Users\seanf'
+++ wslpath -u 'C:\Users\seanf'
++ readonly wHome=/mnt/c/Users/seanf
++ wHome=/mnt/c/Users/seanf
++ readonly CANCELLED=CANCELLED
++ CANCELLED=CANCELLED
++ SetupDir=/usr/local/pengwin-setup.d
++ readonly GOVERSION=1.12
++ GOVERSION=1.12
+ main --debug
++ menu --title 'Tools Menu' --checklist --separate-output 'Install applications or servers\n[SPACE to select, ENTER to confirm]:' 12 70 5 ANSIBLE 'Install tools to deploy Ansible Playbooks' off CLOUDCLI 'Install CLI tools for cloud management' off DOCKER 'Install a secure bridge to Docker Desktop' off HOMEBREW 'Install the Homebrew package manager' off POWERSHELL 'Install PowerShell for Linux' off
++ local menu_choice
+++ whiptail --title 'Tools Menu' --checklist --separate-output 'Install applications or servers\n[SPACE to select, ENTER to confirm]:' 12 70 5 ANSIBLE 'Install tools to deploy Ansible Playbooks' off CLOUDCLI 'Install CLI tools for cloud management' off DOCKER 'Install a secure bridge to Docker Desktop' off HOMEBREW 'Install the Homebrew package manager' off POWERSHELL 'Install PowerShell for Linux' off
++ menu_choice=DOCKER
++ local exit_status=0
++ [[ 0 != 0 ]]
++ [[ -z DOCKER ]]
++ echo DOCKER
+ local menu_choice=DOCKER
+ [[ DOCKER == \C\A\N\C\E\L\L\E\D ]]
+ [[ DOCKER == *\A\N\S\I\B\L\E* ]]
+ [[ DOCKER == *\C\L\O\U\D\C\L\I* ]]
+ [[ DOCKER == *\D\O\C\K\E\R* ]]
+ echo DOCKER
DOCKER
+ bash /usr/local/pengwin-setup.d/docker.sh --debug
Running in debug/verbose mode
++ shift
++ [[ 0 -gt 0 ]]
++ wslpath 'C:\'
+++ cmd-exe /c 'echo %HOMEDRIVE%%HOMEPATH%'
+++ tr -d '\r'
++ readonly 'wHomeWinPath=C:\Users\seanf'
++ wHomeWinPath='C:\Users\seanf'
+++ wslpath -u 'C:\Users\seanf'
++ readonly wHome=/mnt/c/Users/seanf
++ wHome=/mnt/c/Users/seanf
++ readonly CANCELLED=CANCELLED
++ CANCELLED=CANCELLED
++ SetupDir=/usr/local/pengwin-setup.d
++ readonly GOVERSION=1.12
++ GOVERSION=1.12
+ DOCKER_VERSION=18.09.2
+ DOCKER_COMPOSE_VERSION=1.23.2
+ main --debug
+ whiptail --title DOCKER --yesno 'Would you like to install the bridge to Docker?' 8 55
+ echo 'Installing the bridge to Docker.'
Installing the bridge to Docker.
++ docker.exe version
++ grep -c 'docker daemon is not running.\|docker.exe: command not found'
+ local connected=1
+ [[ 1 != 0 ]]
+ whiptail --title DOCKER --yesno 'Docker Desktop appears not to be running, please check it and ensure that it is running correctly. Would you like to try again?' 9 75
+ return
+ [[ DOCKER == *\H\O\M\E\B\R\E\W* ]]
+ [[ DOCKER == *\P\O\W\E\R\S\H\E\L\L* ]]
+ exit_status=0
+ [[ 0 != 0 ]]
+ echo 'Unholding pengwin-base & pengwin-setup'
Unholding pengwin-base & pengwin-setup
+ sudo apt-mark unhold pengwin-base pengwin-setup
+ ByeMessage
+ whiptail --title 'Setup is complete.' --msgbox 'You may run pengwin-setup again any time by typing $ pengwin-setup\n\nYou may open a browser link to get help any time by typing: $ pengwin-help' 8 80
+ exit 0
```

I originally read the above log as saying `docker.exe` was not found (now I'm not sure), so I created a symlink: `ln -s /mnt/c/Program\ Files/Docker/Docker/resources/bin/docker.exe ~/bin/` and that apparently fixed the problem. 

Two weird things though: looking in `$PATH` now, I can see that my Windows path is now in Pengwin's `$PATH`, despite my earlier report of https://github.com/WhitewaterFoundry/Pengwin/issues/434. I think `pengwin-setup` modified `/etc/wsl.conf` when it mounted `C:\` as `/c/`, so I can't work out what's going on now.

Secondly, even if I remove the symlink, `pengwin-setup` actually works with Docker now.

In short, I can't reproduce it now. ¯\\\_(ツ)\_/¯

**Comment by jkstarclub** [comment]
docker >> settings >> wsl2 >> button 1 to right >> button 2 to right
